services/ui-src/src/components/fields/RadioField.test.tsx

```diff
@@ -98,3 +102,37 @@ describe("<RadioField />", () => {
     mockGetValues(undefined);
   });
 });
+
+describe("Radio field hide condition logic", () => {
+  test("Radio field is hidden if its hide conditions' controlling element has a matching answer", async () => {
+    mockGetValues({
+      elements: [
+        {
+          answer: "yes",
+          type: "reportingRadio",
+          label: "Should we hide the other radios on this page?",
+          id: "reporting-radio",
+        },
+      ],
+    });
+    render(RadioFieldComponent);
+    const radioField = screen.queryByText("Choice 1");
+    expect(radioField).toBe(null);
```

Comment:
I'm making all of these suggestions individually so that you can apply them with 1 click from the Github UI, if you like.
```suggestion
    expect(radioField).not.toBeVisible();
```